user.lisp

```diff
@@ -218,10 +218,14 @@ such a case, kill the shell command to resume StumpWM."
 
 (defcommand-alias abort keyboard-quit)
 
+(defvar *quit-prompt* "Really close StumpWM? "
+  "The prompt asked before quitting StumpWM.")
 
 (defcommand quit () ()
-"Quit StumpWM."
-  (throw :top-level :quit))
+  "Quit StumpWM."
+  (when (y-or-n-p *quit-prompt*)
+    #+sbcl (sb-ext:exit)
+    #-sbcl (throw :top-level :quit)))
 
```

Comment:
We don't support anything besides sbcl, so you can get rid of the sbcl hints.  I don't like the use of (sb-ext:exit) because this is the posix extension which implements the 'exit()' call from C which is "quit now, no backsies, sorry if you were in the middle of saving something we're dead now bye".  The way this is currently handled is that it throws a top-level condition that allows stumpwm the opportunity to exit gracefully.  

If you've got a  vanilla mode-line that is causing stumpwm not to exit cleanly then that's a bug, otherwise it sounds like you might have something configured that is causing stumpwm to misbehave.

Reply:
It looks like the mode line configuration (yet again!) has caused stumpwm to misbehave.t